[synthetic] Epoch 1/3:   0%|                                                                                                                                                                                                                                                                 | 0/1062 [00:00<?, ?it/s]Passing a tuple of `past_key_values` is deprecated and will be removed in Transformers v4.48.0. You should pass an instance of `EncoderDecoderCache` instead, e.g. `past_key_values=EncoderDecoderCache.from_legacy_cache(past_key_values)`.
[synthetic] Epoch 1/3:   0%|▏                                                                                                                                                                                                                                             | 1/1062 [00:01<26:09,  1.48s/it, loss=40.7]
Traceback (most recent call last):
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 58, in <module>
    train_loop(train_loader, lr = 1e-3, epochs=num_epochs, name="synthetic")
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 50, in train_loop
    loss = wrapper(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/Desktop/learning-nlp/_modules.py", line 19, in forward
    outputs = self.model(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/transformers/models/t5/modeling_t5.py", line 1934, in forward
    lm_logits = self.lm_head(sequence_output)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 125, in forward
    return F.linear(input, self.weight, self.bias)
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 978.00 MiB. GPU 0 has a total capacity of 11.90 GiB of which 620.12 MiB is free. Including non-PyTorch memory, this process has 11.28 GiB memory in use. Of the allocated memory 10.55 GiB is allocated by PyTorch, and 571.42 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
Traceback (most recent call last):
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 58, in <module>
    train_loop(train_loader, lr = 1e-3, epochs=num_epochs, name="synthetic")
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 50, in train_loop
    loss = wrapper(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/Desktop/learning-nlp/_modules.py", line 19, in forward
    outputs = self.model(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/transformers/models/t5/modeling_t5.py", line 1934, in forward
    lm_logits = self.lm_head(sequence_output)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 125, in forward
    return F.linear(input, self.weight, self.bias)
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 978.00 MiB. GPU 0 has a total capacity of 11.90 GiB of which 620.12 MiB is free. Including non-PyTorch memory, this process has 11.28 GiB memory in use. Of the allocated memory 10.55 GiB is allocated by PyTorch, and 571.42 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
